scalapb-runtime/shared/src/main/scala/scalapb/StructUtils.scala

```diff
@@ -0,0 +1,122 @@
+package scalapb
+
+import java.util.Base64
+
+import com.google.protobuf.ByteString
+import com.google.protobuf.struct.Value.Kind
+import com.google.protobuf.struct.{ListValue, Struct, Value}
+import scalapb.descriptors._
+
+import scala.util.{Failure, Success, Try}
+
+object StructUtils {
+  case class StructDeserError(index: Int, error: String)
```

Comment:
Let's rename to `StructParsingError`. I think that having the index is not particularly useful since that single number doesn't help understand where the problem happened. I suggest that we use the `FieldDescriptor` instead.